oss_src/sframe/groupby_aggregate_operators.hpp

```diff
@@ -1255,9 +1256,107 @@ class count_distinct: public group_aggregate_value {
     iarc >> m_values;
   }
 
- private:
+ protected:
   std::unordered_set<flexible_type> m_values;
 };
+
+/**
+ * Implements an aggregator that keeps track of the unique elements
+ */
+class distinct: public count_distinct {
+ public:
+  
+  group_aggregate_value* new_instance() const override {
+    distinct* ret = new distinct;
+    return ret;
+  }
+
+  /// Emits the zip result
+  flexible_type emit() const override {
+    flex_list ret(m_values.size());
+    size_t i = 0;
+    for (const auto& k: m_values) {
+      ret[i] = k;
+      i++;
+    }
+    return ret;
+  }
+  
+  flex_type_enum set_input_type(flex_type_enum type) override {
+    return flex_type_enum::LIST;
+  }
+
+  /// Name of the class
+  std::string name() const override {
+    return "Distinct";
+  }
+
+};
+
+/**
+ * Implements an aggregator that computes frequncies for each unique value. 
+ */
+class frequency_count: public group_aggregate_value {
+ public:
+  /// Returns a new empty instance of sum with the same type
+  group_aggregate_value* new_instance() const {
+    frequency_count* ret = new frequency_count;
+    return ret;
+  }
+
+  void add_element_simple(const flexible_type& flex) {
+    if (m_values.find(flex) == m_values.end()) {
+      m_values[flex] = 1;
+    } else {
+      m_values[flex] += 1;
```

Comment:
This ends up performing 2 lookups. once for find() and once for the map indexing.

Since the map value is guaranteed to zero-initialize when value type is an integer, all this code has to do is only:
```
m_values[flex] += 1;
```